### ISSUE TITLE
board_permissions_view has no repair tools

### ISSUE THREAD

**Issue by jdarwood007** [SimpleMachines/SMF#7503]
#### Description
If board_permissions_view gets corrupted or is out of sync, there is no maintenance tools.  There is no find and repair tasks.
We need tools to correct this when it gets broken.  When this gets out of sync, users may not be able to view boards, despite being in the correct groups.

**Comment by jdarwood007** [comment]
@albertlast Since you added the initial implantation here, would you like to take this on?

**Comment by albertlast** [comment]
this stuff get rebuild when a change happen right?

**Comment by jdarwood007** [comment]
Correct, but assume somehow that didn't happen or if the table got corrupted.  We have all the information in the other table, but no way to detect or repair the corrupt logic.

I came up with this logic for mysql to detect a mismatch.  I know postgresql doesn't do group_concat, but there are other methods.  I just haven't came up with said query.
```
					SELECT
						b.id_board,
						b.member_groups,
						pv.mbrgrp
					FROM {db_prefix}boards AS b
					LEFT JOIN (
						SELECT
							xpv.id_board, 
							GROUP_CONCAT(xpv.id_group ORDER BY xpv.id_group) AS mbrgrp
						FROM {db_prefix}board_permissions_view AS xpv
							GROUP BY xpv.id_board
						) AS pv ON pv.id_board = b.id_board
					WHERE 
						pv.id_board IS NULL
						OR b.member_groups != pv.mbrgrp
```

**Comment by albertlast** [comment]
``GROUP_CONCAT(xpv.id_group ORDER BY xpv.id_group)`` = ``string_agg(xpv.id_group, ',' ORDER BY xpv.id_group)``
this query could be very slow,
so question: would be better instead of doing a check, just run rebuild on regular basis?

**Comment by jdarwood007** [comment]
I would say it should go into find and repair errors.  If admins are having issues with permissions after something goes wrong the find and repair errors typically fixes oddities like this.

Yea I realize it could be slow.  I initially had it as a lateral, but I think this would be faster.

**Comment by albertlast** [comment]
in general when it find and error than it should rebuild complet,
this is the default behavior of smf on changes and this should be also the same for repair,
all other approache are to smart and src for issues,

**Comment by sbulen** [comment]
Was there a problem or a bug here?  Spontaneous corruption isn't exactly common...  

I'd say that if we haven't seen issues, to leave it alone.
